File: src/api/MoRS.ManagementSystem.Application/DTOs/User/UpdateUserRequest.cs
using System.ComponentModel.DataAnnotations;

namespace MoRS.ManagementSystem.Application.DTOs.User;

public class UpdateUserRequest
{
    [Required]
    [StringLength(50, MinimumLength = 2)]
    public required string Name { get; set; }

    [Required]
    [StringLength(50, MinimumLength = 2)]
    public required string Surname { get; set; }

    [Required]
    [EmailAddress]
    [RegularExpression(@"^[^@\s]+@[^@\s]+\.[^@\s]+$")]
    public required string Email { get; set; }
    [Required]
    [RegularExpression(@"^06[0-9]{7,8}$", ErrorMessage = "Broj telefona mora počinjati sa 06 i imati ukupno 9 ili 10 cifara bez razmaka i specijalnih znakova.")]
    public required string PhoneNumber { get; set; }

    [Required]
    public bool IsRestricted { get; set; }

    [Key]
    public int RoleId { get; set; }
}

File: src/mobile/lib/models/user/update_user_request.dart
class UpdateUserRequest {
  final String name;
  final String surname;
  final String email;
  final String phoneNumber;
  final bool isRestricted;
  final int roleId;

  UpdateUserRequest({
    required this.name,
    required this.surname,
    required this.email,
    required this.phoneNumber,
    required this.isRestricted,
    required this.roleId,
  });

  factory UpdateUserRequest.fromJson(Map<String, dynamic> json) {
    return UpdateUserRequest(
      name: json['name'] as String,
      surname: json['surname'] as String,
      email: json['email'] as String,
      phoneNumber: json['phoneNumber'] as String,
      isRestricted: json['isRestricted'] as bool,
      roleId: json['roleId'] as int,
    );
  }
  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'surname': surname,
      'email': email,
      'phoneNumber': phoneNumber,
      'isRestricted': isRestricted,
      'roleId': roleId,
    };
  }

  @override
  String toString() {
    return 'UpdateUserRequest(name: $name, surname: $surname, email: $email, phoneNumber: $phoneNumber, isRestricted: $isRestricted, roleId: $roleId)';
  }
}
